docker-compose-rule-core/src/main/java/com/palantir/docker/compose/execution/DefaultDockerCompose.java

```diff
@@ -45,102 +47,111 @@
     private static final Duration LOG_TIMEOUT = standardMinutes(1);
     private static final Logger log = LoggerFactory.getLogger(DefaultDockerCompose.class);
 
-    private final Command command;
+    private final Command dockerComposeCommand;
+    private final DockerComposeExecutable dockerComposeRawExecutable;
+    private final Command dockerCommand;
     private final DockerMachine dockerMachine;
-    private final DockerComposeExecutable rawExecutable;
 
 
     public DefaultDockerCompose(DockerComposeFiles dockerComposeFiles, DockerMachine dockerMachine, ProjectName projectName) {
         this(DockerComposeExecutable.builder()
                 .dockerComposeFiles(dockerComposeFiles)
                 .dockerConfiguration(dockerMachine)
                 .projectName(projectName)
-                .build(), dockerMachine);
+                .build(),
+                DockerExecutable.builder()
+                        .dockerConfiguration(dockerMachine)
+                .build(),
+                dockerMachine);
     }
 
-    public DefaultDockerCompose(DockerComposeExecutable rawExecutable, DockerMachine dockerMachine) {
-        this.rawExecutable = rawExecutable;
-        this.command = new Command(rawExecutable, log::trace);
+    public DefaultDockerCompose(
+            DockerComposeExecutable dockerComposeRawExecutable,
+            DockerExecutable dockerRawExecutable,
```

Comment:
This also seems safe from a brief github search (I think 1 person has used it externally and none internally)

Reply:
Yeah it seems reasonable enough, will add comments about break-impact next time!